### ISSUE TITLE
Documentation around getting posts from subreddit does not make sense

### ISSUE THREAD

**Issue by lloydjatkinson** [sirkris/Reddit.NET#107]
```csharp
redditClient.Subreddit("programming").Posts.GetBest();
```

The above code returns "best" posts across all of Reddit, not /r/programming. I believe the libraries API is too confusing and does not work as intended.

```csharp
redditClient.Subreddit("programming").Posts.Best;
```

The above code has exactly the same problem. I am not sure why there is both a property *and* a method, if this is some caching feature then why not make it a method and allow the consumer to not worry about it. Either way, the intellisense documentation doesn't explain it.

Looking further down the README there is a second approach described showing the use of "Models" (?).

```csharp
// Get the top post from a subreddit.
var topPost = r.Models.Listings.Top(new TimedCatSrListingInput(), "AskReddit").JSON.Data.Things[0].Data;
```

1. What is a "model"? Do you mean in the DTO/Model sense? If so, why does the example code show you having to manually navigate the JSON, if it was a model I'd expect it to be deserialized already? JSON isn't a property anymore either, I assume the docs were not updated to reflect this.
2. What on earth is `TimedCatSrListingInput`? Why do I even need that to search for a subreddit?
3. Why are there seemingly two radically different approaches to getting the posts? The first approach seemingly ignores the subreddit argument string, and the second approach only gives you half the information - it's not a `LinkPost` so you can't get the permalink URL.

**Comment by FireCubeStudios** [comment]
Getting posts via “best” will always get posts from the front page. In the docs the example is there when getting posts from the front page. Instead you should get posts with “new” or “top” etc.

**Comment by FireCubeStudios** [comment]
Also the “timedcatsrlisting” is used only when getting “Top” posts. It is used to know if you want top posts of the day or week or all time etc.

I’m not the creator or expert in the api but that’s what I have used it for also for the models you can ignore those

**Comment by sirkris** [comment]
This can be a bit confusing, I suppose.  I'll see about adding some mention of this to the docs in the next release.

**Comment by sirkris** [comment]
> I am not sure why there is both a property and a method,

The method allows you to change the query parameters, like setting a different limit.  For example:

GetBest(limit:10) would query for 10 posts instead of the default.

The property is for caching, as you guessed.  The method should only be used if you're looking to change the query parameters or just want to clear the cache.

**Comment by sirkris** [comment]
The more I think about it, the more I question whether Best should even be included in the subreddit controller at all.  Not sure why I put it there initially, tbh.  The API docs group that endpoint together with Hot/New/etc so I probably just didn't notice at the time that one of them didn't require a specified subreddit.

The problem at this point is BC, so outright removing it in this next release is out of the question.  But I am beginning to lean towards a deprecation flag.

**Comment by sirkris** [comment]
This has been resolved by #125.  Please feel free to re-open if you're still having issues.
